src/psyclone/transformations.py

```diff
@@ -434,6 +434,10 @@ def validate(self, node, options=None):
         # Check that the kernel does not access any data or routines via a
         # module 'use' statement or that are not captured by the SymbolTable
         for candidate in node.walk((Reference, CodeBlock)):
```

Comment:
This makes me realise two things:

 1. this validate should really be shared with the `ACCRoutineTrans` class;
 2. we should also check for any `Calls` because we don't yet support recursively adding directives to called routines (and anyway, the performance tends to be rubbish).

Reply:
I agree with 1) . This is subjective but a common Transformation ancestor felt forced for OMP and ACC, so I went the Mixin route, let me know what you think.
I subtly disagree with 2), I think it should follow the caller "is_available_on_device" info. But in practice is the same as not accepting Calls since its currently hardcoded to False. But it will allow to expand this on the future and can properly filter Intrinsics (which currently were only failing at code generation time for both directives!).

Reply:
Thanks Sergi. I like the Mixin solution.